pi_ldapproxy/preamblecache.py

```diff
@@ -0,0 +1,50 @@
+from twisted.internet import reactor
+from twisted.logger import Logger
+
+log = Logger()
+
+class PreambleCache(object):
+    # (see http://twistedmatrix.com/documents/current/core/howto/trial.html)
+    callLater = reactor.callLater
+
+    def __init__(self, timeout):
+        self.timeout = timeout
+
+        #: Map of dn to tuples (app marker, insertion timestamp)
+        self._preambles = {}
```

Comment:
So the cache is only contained in the dictionary?
What about different processes and different threads? 
We would not share the cache between different processes or even different machines. 
Maybe we should use something like memcached or redis? Is this a huge overhead?